src/measure_theory/decomposition/lebesgue.lean

```diff
@@ -686,4 +731,284 @@ end⟩
+  {t : signed_measure α} {f : α → ℝ} (hf : measurable f) (htμ : t ⊥ᵥ μ.to_ennreal_vector_measure) :
+  t.to_jordan_decomposition.pos_part + μ.with_density (λ (x : α), ennreal.of_real (f x)) ⊥ₘ
+  t.to_jordan_decomposition.neg_part + μ.with_density (λ (x : α), ennreal.of_real (-f x)) :=
+begin
+  rw [mutually_singular_ennreal_iff, total_variation_mutually_singular_iff] at htμ,
+  change _ ⊥ₘ vector_measure.equiv_measure.to_fun (vector_measure.equiv_measure.inv_fun μ) ∧
+         _ ⊥ₘ vector_measure.equiv_measure.to_fun (vector_measure.equiv_measure.inv_fun μ) at htμ,
+  rw [vector_measure.equiv_measure.right_inv] at htμ,
+  exact ((jordan_decomposition.mutually_singular _).symm.add
+    (htμ.1.symm.of_absolutely_continuous (with_density_absolutely_continuous _ _))).symm.add
+    ((htμ.2.symm.of_absolutely_continuous (with_density_absolutely_continuous _ _)).symm.add
+      (with_density_of_real_mutually_singular hf).symm).symm
+end
+
+lemma to_jordan_decomposition_eq_of_eq_add_with_density
+  {t : signed_measure α} {f : α → ℝ} (hf : measurable f) (hfi : integrable f μ)
+  (htμ : t ⊥ᵥ μ.to_ennreal_vector_measure) (hadd : s = t + μ.with_densityᵥ f) :
+  s.to_jordan_decomposition = @jordan_decomposition.mk α _
+    (t.to_jordan_decomposition.pos_part + μ.with_density (λ x, ennreal.of_real (f x)))
+    (t.to_jordan_decomposition.neg_part + μ.with_density (λ x, ennreal.of_real (- f x)))
+    (by { haveI := is_finite_measure_with_density_of_real hfi.2, apply_instance })
+    (by { haveI := is_finite_measure_with_density_of_real hfi.neg.2, apply_instance })
+    (jordan_decomposition_add_with_density_mutually_singular hf htμ) :=
+begin
+  haveI := is_finite_measure_with_density_of_real hfi.2,
+  haveI := is_finite_measure_with_density_of_real hfi.neg.2,
+  refine to_jordan_decomposition_eq _,
+  simp_rw [jordan_decomposition.to_signed_measure, hadd],
+  ext i hi,
+  rw [vector_measure.sub_apply, to_signed_measure_apply_measurable hi,
+      to_signed_measure_apply_measurable hi, add_apply, add_apply,
+      ennreal.to_real_add, ennreal.to_real_add, add_sub_comm,
+      ← to_signed_measure_apply_measurable hi, ← to_signed_measure_apply_measurable hi,
+      ← vector_measure.sub_apply, ← jordan_decomposition.to_signed_measure,
+      to_signed_measure_to_jordan_decomposition, vector_measure.add_apply,
+      ← to_signed_measure_apply_measurable hi, ← to_signed_measure_apply_measurable hi,
+      with_densityᵥ_eq_with_density_pos_part_sub_with_density_neg_part hfi,
+      vector_measure.sub_apply];
+  exact (measure_lt_top _ _).ne
+end
+
+/-- Given a measure `μ`, signed measures `s` and `t`, and a function `f` such that `t` is
+mutually singular with respect to `μ` and `s = t + μ.with_densityᵥ f`, we have
+`t = singular_part s μ`, i.e. `t` is the singular part of the Lebesgue decomposition between
+`s` and `μ`. -/
+theorem eq_singular_part
+  {t : signed_measure α} {f : α → ℝ} (hf : measurable f) (hfi : integrable f μ)
```

Comment:
I'm not sure if I understand. In the case that `f` is not integrable, we have `s = t`, but I'm not sure how to show `t = s.singular_part μ` from there

Reply:
Don't you have `t = t.singular_part µ` under the hypothesis `htµ`?

Reply:
Can you elaborate? I don't quite understand what you mean

Reply:
I don't really know the actual definitions of the objects used here, so I am only making an informal guess, but since `t` is supposed to be mutually singular with `µ`, it should be equal to its own singular part wrt `µ`? Then you have `t = t.singular_part µ` and `t = s` as you explained, from which you get `t = s.singular_part µ`.

But again, maybe the first step is nonsense since I did not look at those objects closely.

Reply:
I see. I was stuck on the fact that we will need `eq_singular_part` to prove what you just said, but that makes a lot of sense since 0 is integrable.